### PR TITLE
feat (ui/react): introduce experimental_throttle in useChat, useCompletion

### PR DESCRIPTION
**Pull request by lgrammel**
## Summary
Introduces an `experimental_throttle` setting on `useChat` and `useCompletion` that allows passing in a custom throttle wait in ms (React only).

## Background
By default, the UI is re-rendered on every chunk that arrives. This can overload the rendering, especially on slower devices or when complex components need updating (e.g. Markdown). Throttling can mitigate this.

## Related Issues
#1610 

## Tasks

- [x] implementation
- [x] examples
- [x] docs: useCompletion guide
- [x] docs: useCompletion reference
- [x] docs: chatbot guide
- [x] docs: useChat reference
- [x] docs: troubleshooting page
- [x] changeset

### CODE REVIEW

File: content/docs/03-ai-sdk-core/55-testing.mdx [hunk 1]
```diff
@@ -149,8 +149,8 @@ export async function POST(req: Request) {
         `0:"This"\n`,
         `0:" is an"\n`,
         `0:"example."\n`,
-        `d:{"finishMessage":"unknown","usage":{"promptTokens":20,"completionTokens":50},"isContinued":false}`,
-        `d:{"finishMessage":"unknown","usage":{"promptTokens":20,"completionTokens":50},"isContinued":false}`,
+        `e:{"finishReason":"stop","usage":{"promptTokens":20,"completionTokens":50},"isContinued":false}\n`,
+        `d:{"finishReason":"stop","usage":{"promptTokens":20,"completionTokens":50}}\n`,
```
Comment (lgrammel) [inline content/docs/03-ai-sdk-core/55-testing.mdx:153]: unrelated docs fix

File: examples/next-openai/app/api/use-completion-long-response/route.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export async function POST(req: Request) {
       `Write a book in Markdown format. ` +
       `First write a table of contents. ` +
       `Then write each chapter. ` +
-      `Each chapter test MUST HAVE at least 1000 words.`,
+      `Each chapter MUST HAVE at least 1000 words.`,
```
Comment (lgrammel) [inline examples/next-openai/app/api/use-completion-long-response/route.ts:19]: unrelated fix

**Review by jeremyphilemon — APPROVED** [review]
